scripts/tests/chiptest/linux.py

```diff
@@ -0,0 +1,140 @@
+
+    logging.debug("Making / private")
+    if os.system("mount --make-private /") != 0:
+        logging.error("Failed to make / private")
+        logging.error("Are you using --privileged if running in docker?")
+        sys.exit(1)
+
+    logging.debug("Remounting /run")
+    if os.system("mount -t tmpfs tmpfs /run") != 0:
+        logging.error("Failed to mount /run as a temporary filesystem")
+        logging.error("Are you using --privileged if running in docker?")
+        sys.exit(1)
+
+
+def CreateNamespacesForAppTest():
+    """
+    Creates appropriate namespaces for a tool and app binaries in a simulated
+    isolated network.
+    """
+    COMMANDS = [
+        # 2 virtual hosts: for app and for the tool
+        "ip netns add app",
+        "ip netns add tool",
+
+        # create links for switch to net connections
+        "ip link add eth-app type veth peer name eth-app-switch",
+        "ip link add eth-tool type veth peer name eth-tool-switch",
+
+        # link the connections together
+        "ip link set eth-app netns app",
+        "ip link set eth-tool netns tool",
+
+        "ip link add name br1 type bridge",
+        "ip link set br1 up",
+        "ip link set eth-app-switch master br1",
+        "ip link set eth-tool-switch master br1",
+
+        # mark connections up
+        "ip netns exec app ip addr add 10.10.10.1/24 dev eth-app",
+        "ip netns exec app ip link set dev eth-app up",
+        "ip netns exec app ip link set dev lo up",
+        "ip link set dev eth-app-switch up",
+
+        "ip netns exec tool ip addr add 10.10.10.2/24 dev eth-tool",
+        "ip netns exec tool ip link set dev eth-tool up",
+        "ip netns exec tool ip link set dev lo up",
+        "ip link set dev eth-tool-switch up",
+
+        # Force IPv6 to use ULAs that we control
+        "ip netns exec tool ip -6 addr flush eth-tool",
+        "ip netns exec app ip -6 addr flush eth-app",
+        "ip netns exec tool ip -6 a add fd00:0:1:1::2/64 dev eth-tool",
+        "ip netns exec app ip -6 a add fd00:0:1:1::3/64 dev eth-app",
+    ]
+
+    for command in COMMANDS:
+        logging.debug("Executing '%s'" % command)
+        if os.system(command) != 0:
+            logging.error("Failed to execute '%s'" % command)
+            logging.error("Are you using --privileged if running in docker?")
+            sys.exit(1)
+
+    # IPv6 does Duplicate Address  Detection even though
+    # we know ULAs provided are isolated. Wait for 'tenative' address to be gone
+
+    logging.info('Waiting for IPv6 DaD to complete (no tentative addresses)')
+    for i in range(100):  # wait at most 10 seconds
+        output = subprocess.check_output(['ip', 'addr'])
+        if b'tentative' not in output:
+            break
+        time.sleep(0.1)
+
+    if b'tentative' in output:
+        logging.warn("Some addresses look to still be tentative")
+    else:
+        logging.info('No more tentative addresses')
+
```

Comment:
nit: this can use [for/else statement](https://book.pythontips.com/en/latest/for_-_else.html#else-clause)
```suggestion
    logging.info('Waiting for IPv6 DaD to complete (no tentative addresses)')
    for i in range(100):  # wait at most 10 seconds
        output = subprocess.check_output(['ip', 'addr'])
        if b'tentative' not in output:
            logging.info('No more tentative addresses')
            break
        time.sleep(0.1)
    else:
        logging.warn("Some addresses look to still be tentative")
```

Reply:
Neat - did not know about the else option. TIL. Thanks!